### PR TITLE
added PATCH /basket/checkout endpoint

### PR DESCRIPTION
**Pull request by nabeelamirch**
added PATCH /basket/checkout endpoint

resolves: PCI-605

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/orders/api/controller/BasketController.java [hunk 1]
```diff
@@ -113,7 +115,12 @@ public BasketController(final BasketMapper mapper,
         Checkout checkout = checkoutService.createCheckout(item, EricHeaderHelper.getIdentity(request));
         trace("Successfully created checkout with id "+checkout.getId(), requestId);
 
-        return ResponseEntity.status(HttpStatus.OK).body(null);
+        return ResponseEntity.status(HttpStatus.OK).body(checkout);
+    }
+
+    @PatchMapping("${uk.gov.companieshouse.orders.api.basket.payment}/{id}")
+    public ResponseEntity<?> patchBasketPaymentDetails(final @PathVariable String id) {
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/orders/api/controller/BasketController.java]: These seem to be missing:
* Request body
* Request header?

Comment (nabeelamirch) [reply src/main/java/uk/gov/companieshouse/orders/api/controller/BasketController.java]: I will add the body and header in a future story, however I will change the code to allow application/merge-patch.json

File: src/main/java/uk/gov/companieshouse/orders/api/controller/BasketController.java [hunk 2]
```diff
@@ -113,7 +116,16 @@ public BasketController(final BasketMapper mapper,
         Checkout checkout = checkoutService.createCheckout(item, EricHeaderHelper.getIdentity(request));
         trace("Successfully created checkout with id "+checkout.getId(), requestId);
 
-        return ResponseEntity.status(HttpStatus.OK).body(null);
+        return ResponseEntity.status(HttpStatus.OK).body(checkout);
+    }
+
+    @PatchMapping(path = "${uk.gov.companieshouse.orders.api.basket.payment}/{id}",
+            consumes = "application/merge-patch+json")
+    public ResponseEntity<String> patchBasketPaymentDetails(final @RequestBody BasketPaymentRequestDTO basketPaymentRequestDTO,
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/orders/api/controller/BasketController.java]: I have not tried using `application/merge-patch+json` with a BasketPaymentRequestDTO payload. In the Orders API, the request body as a `JsonMergePatch`.

File: src/main/java/uk/gov/companieshouse/orders/api/controller/BasketController.java [hunk 3]
```diff
@@ -113,7 +116,16 @@ public BasketController(final BasketMapper mapper,
...
+                                                            final @PathVariable String id,
+                                                            final @RequestHeader(REQUEST_ID_HEADER_NAME) String requestId) {
+        trace("ENTERING updateCertificateItem(" + basketPaymentRequestDTO + ", " + id + ", " + requestId + ")", requestId);
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/orders/api/controller/BasketController.java]: Trace message gives wrong method name?

Comment (nabeelamirch) [reply src/main/java/uk/gov/companieshouse/orders/api/controller/BasketController.java]: Will update now

File: src/main/java/uk/gov/companieshouse/orders/api/controller/BasketController.java (hunk 2 above)
Comment (nabeelamirch) [reply src/main/java/uk/gov/companieshouse/orders/api/controller/BasketController.java]: Will change this back to use application/json

File: src/main/java/uk/gov/companieshouse/orders/api/dto/BasketPaymentRequestDTO.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package uk.gov.companieshouse.orders.api.dto;
+
+public class BasketPaymentRequestDTO {
+
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/orders/api/dto/BasketPaymentRequestDTO.java]: Given that this is not a merge patch, I assume we will need attributes in the DTO for each of the fields as spec'ed in the API. 😄

Comment (nabeelamirch) [reply src/main/java/uk/gov/companieshouse/orders/api/dto/BasketPaymentRequestDTO.java]: I will add a simplified version of these in now.

**Pull request merged**